### PR TITLE
[9.0][FIX][purchase_request_to_rfq] taxes not determined in the PO line.

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Solves https://github.com/OCA/purchase-workflow/issues/344.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@adrienpeiffer @JonathanNEMRY This is an important improvement.

**Review by JordiBForgeFlow — APPROVED** [review]

**Review by lasley — COMMENTED** [review]
Code looks good. Can you please update the test to include logic for taxes?

**Comment by adrienpeiffer** [comment]
Hi,
I'm not sure I understand the problem ...
Is there a problem when you create a purchase order from scratch (Without PR).
If there is no problem, the solution is directly on the purhase module and it should not need all this code

**Comment by JordiBForgeFlow** [comment]
@adrienpeiffer The problem is that when you create a PO from a PR, and the supplier has a price agreement (supplier prices defined in the product), these are not considered. Same with taxes.

**Comment by adrienpeiffer** [comment]
BTW during the migration on version 10.0 I modified this method of preparation of value. This fixed several problems on UOM, price and other things. You can view the commits from the migration. There may be things to backport if you want.

**Comment by JordiBForgeFlow** [comment]
@adrienpeiffer OK, we will then review in 10.0. Thanks!

**Comment by adrienpeiffer** [comment]
Especially, I launch onchange directly on a memory record.

File: purchase_request_to_rfq/wizard/purchase_request_line_make_purchase_order.py [hunk 1]
```diff
@@ -138,37 +135,40 @@ def _prepare_purchase_order_line(self, po, item):
...
+            date=po.date_order and po.date_order[:10],
+            uom_id=item.product_id.uom_po_id)
+        price_unit = self.env['account.tax']._fix_tax_included_price(
+            seller.price, item.product_id.supplier_taxes_id, taxes_id) \
+            if seller else 0.0
+        if (price_unit and seller and po.currency_id and
+                seller.currency_id != po.currency_id):
+            price_unit = seller.currency_id.compute(
+                price_unit, po.currency_id)
         vals = po_line_obj.onchange_product_id()
```
Comment (adrienpeiffer) [inline purchase_request_to_rfq/wizard/purchase_request_line_make_purchase_order.py]: So, I think the problem is the onchange isn't launched on a memory record.
See https://github.com/odoo/odoo/blob/9ec95b2a682ca885c0796c8e299ac720e8b7e5d8/addons/purchase/models/purchase.py#L748
I fix that in my PR for the migration to 10.0

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]

File: purchase_request_to_rfq/wizard/purchase_request_line_make_purchase_order.py (hunk 1 above)
Comment (JordiBForgeFlow) [reply purchase_request_to_rfq/wizard/purchase_request_line_make_purchase_order.py]: Thanks for pointing this out!

**Review by AaronHForgeFlow — APPROVED** [review]
Code review + functional test. :white_check_mark:

**Comment by adrienpeiffer** [comment]
@aheficent As I mentionned above. It's better to call the onchange on purchase order line properly. It should make the job.

**Comment by LoisRForgeFlow** [comment]
@adrienpeiffer what do you mean? We added your changes, or am I wrong?  :confused:

**Comment by MiquelRForgeFlow** [comment]
Now https://github.com/OCA/purchase-workflow/pull/349 is full added here.

**Comment by JordiBForgeFlow** [comment]
Superseded by https://github.com/OCA/purchase-workflow/pull/350

**Pull request closed without merging**